Sources/InstantSearchCore/FacetList/FacetListInteractor+Controller.swift

```diff
@@ -29,18 +29,20 @@ extension FacetList {
 
     public func connect() {
 
-      setControllerItemsWith(facets: facetListInteractor.items, selections: facetListInteractor.selections, controller: controller, presenter: presenter)
-
+      ControllerConnection.setControllerItemsWith(facets: facetListInteractor.items, selections: facetListInteractor.selections, controller: controller, presenter: presenter)
+      
       controller.onClick = { [weak facetListInteractor] facet in
         facetListInteractor?.computeSelections(selectingItemForKey: facet.value)
       }
 
-      facetListInteractor.onItemsChanged.subscribePast(with: controller) { controller, facets in
-        self.setControllerItemsWith(facets: facets, selections: self.facetListInteractor.selections, controller: controller, presenter: self.presenter)
+      facetListInteractor.onItemsChanged.subscribePast(with: controller) { [weak interactor = self.facetListInteractor, presenter = self.presenter] controller, facets in
```

Comment:
Because `SelectableListPresentable` is not a class protocol. Anyways, presenter provides one simple function and shouldn't be a source of a reference cycle.

Reply:
yep agreed